Design/trivially_exportable_citations.md

```diff
@@ -0,0 +1,138 @@
+  - if none of `doi`, `bibtex` or `endnote` are provided, `url` must be provided (there must be something there, even if this isn't citable a URL is better than nothing)
+
+Some helper "toString" methods may be beneficial.
+
+### Algorithm citations
+
+An additional `static` method will be added to `IAlgorithm` that returns a `std::vector<Citation const> const`.
+The developer has the option to overload this if the algorithm has relevant citations.
+By default the implementation in `IAlgorithm` will return an empty vector.
+
+### Citation recording
+
+- `WorkspaceHistory` and `AlgorithmHistory` shall be extended to include storage for framework and algorithm level citations respectively
+  - a `std::vector<Citation>` is likely sufficient
+- The requisite IO methods will be extended to handle loading and saving these citations to NeXus
+  - `WorkspaceHistory::loadNexus`
+  - `WorkspaceHistory::saveNexus`
+  - `AlgorithmHistory::saveNexus`
+
+Citations will be recorded in NeXus files using the [`NXcite`](http://download.nexusformat.org/doc/html/classes/base_classes/NXcite.html#nxcite) class.
+
+### Citation exporting
+
+There will be functionality to generate plain text lists of citations in all supported formats.
+This will be similar to the way we currently generate Python scripts form algorithm histories.
+
+A new algorithm `ExportCitations` shall be implemented.
+This algorithm takes a workspace as a property, gives the option of the desired export option and will output a file.
+
+The Algorithm History GUI may be extended to provide this functionality in a similar manner to Python scripts generated from the algorithm histories.
+
+### Documentation
+
+There are two main requirements for documentation integration:
+  - Listing the citations in the algorithm documentation page
+  - Allowing the documentation writer (developer) to cite the citations at specific points in the documentation
+
+The first point is trivial as we already do similar things for properties, related algorithms, etc.
+
+The second will involve more work.
+I don't have enough Sphinx know-how to suggest how this may be implemented.
+
+## Design considerations
+
+### Property dependant citations
+
+I briefly considered making the citations returned for algorithms dependant on the algorithm being run.
+This allows the algorithm to record what citations it used during execution.
+
+The advantage this gives is greater accuracy of citations (e.g. toggling an option may divert control flow away from a particular citable method).
+
+It was chosen not to go down this route for the following reasons:
+  - It adds complexity
+  - It adds runtime dependency
+    - This prevents tooling for algorithm documentation
+
+### Backwards compatibility
+
+What happens when loading old workspaces with no citations in the NeXus file?
+
+### `constexpr`
+
+Is it worth using `char *` as the string format to be able to `constexpr` the `Citation` structure?
```

Comment:
Maybe better to work out at implementation if we find places that benefit from the information at compile time? I can see though that a lot will just be static strings...